### DISCUSSION TITLE
Limit registration to known domains

### DISCUSSION THREAD

**Discussion by Parodontitis** [zitadel/zitadel#8673] [Q&A] [answered]
Hi Sirs, I would like our users to be able to register themselves. However, the registration should be limited to email addresses from our three domains (xyz.org, xyz.com and abcd.org).

At the moment I am blind and can't find the solution. Who could help me?

**Comment by hifabienne** [comment, marked as answer]
Hi @Parodontitis 

You can use domain discovery so when a user enters the username user@xyz.org, they will be redirected to the organization where this domain is added. [https://zitadel.com/docs/guides/solution-scenarios/domain-discovery](https://zitadel.com/docs/guides/solution-scenarios/domain-discovery)
Additionally you can add a ZITADEL action which is a javascript snippet to the pre creation flow to check if the username does include that domain. https://zitadel.com/docs/apis/actions/internal-authentication#pre-creation
